resolwe_bio/processes/samtools/samtools_coverage.py

```diff
@@ -0,0 +1,162 @@
+            description="Select BAM file(s) for the analysis. Coverage information will "
+            "be calculated from the merged alignments.",
+        )
+        region = StringField(
+            label="Region",
+            description="Regions can be specified as: RNAME:STARTPOS-ENDPOS "
+            "and all position coordinates are 1-based, where RNAME is the "
+            "name of the contig. If the input BAM file was generated by "
+            "General RNA-seq pipeline, you should use only chromosome "
+            "numbers to subset the input file, e.g. 3:30293-39103.",
+            required=False,
+        )
+
+        class AdvancedOptions:
+            """Advanced options."""
+
+            min_read_length = IntegerField(
+                label="Minimum read length",
+                required=False,
+                description="Ignore reads shorter than specified number of "
+                "base pairs",
+            )
+            min_mq = IntegerField(
+                label="Minimum mapping quality",
+                required=False,
+                description="Minimum mapping quality for an alignment to be used.",
+            )
+            min_bq = IntegerField(
+                label="Minimum base quality",
+                required=False,
+                description="Minimum base quality for a base to be considered.",
+            )
+            excl_flags = StringField(
+                label="Filter flags",
+                required=False,
+                description="Filter flags: skip reads with mask bits set "
+                "[UNMAP,SECONDARY,QCFAIL,DUP]",
+            )
+            depth = IntegerField(
+                label="Maximum allowed coverage depth",
+                required=False,
+                description="If 0, depth is set to the maximum integer value "
+                "effectively removing any depth limit.",
+            )
+            no_header = BooleanField(
+                label="No header",
+                default=False,
+                description="Don't print a header in tabular mode.",
+            )
+
+        advanced = GroupField(AdvancedOptions, label="Advanced options")
+
+    class Output:
+        """Output fields for SamtoolsCoverage."""
+
+        table = FileField(label="Output coverage table")
+        build = StringField(label="Build")
+        species = StringField(label="Species")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        output_name = "samtools_coverage.tsv"
+
+        num_of_bams = len(inputs.bam)
+        if num_of_bams > 1:
+            for i in range(1, num_of_bams):
+                if inputs.bam[0].output.species != inputs.bam[i].output.species:
+                    self.error("Not all BAM files are from the same species.")
+
+                if inputs.bam[0].output.build != inputs.bam[i].output.build:
+                    self.error("Not all BAM files have the same genome build.")
```

Comment:
Could be simplified to:

```
for bam in inputs.bam:
    if inputs.bam[0].output.species != bam.output.species:
        self.error("Not all BAM files are from the same species.")

    if inputs.bam[0].output.build != bam.output.build:
        self.error("Not all BAM files have the same genome build.")
```

Perhaps it could also say which species / build are mismatched.